web/src/helpers/getPageContent.ts

```diff
@@ -12,19 +12,42 @@ export async function getPageContent(slug: string[]) {
 		},
 	};
 	const query = qs.stringify(params, { addQueryPrefix: true });
-	const respose = await fetch(`${process.env.CMS_API}/menus/1${query}`, {
-		// headers: {
-		//     authorization: `Bearer ${process.env.CMS_TOKEN}`,
-		// },
-	});
-	const data = (await respose.json()).data;
-	const page = data.attributes.items.data.find((item: { attributes: { url: string } }) => item.attributes.url.endsWith(slug[0]));
-	const pageRelation = page?.attributes?.page_relation.data;
-
-	let pageContent;
-	if (pageRelation) {
+	let response, data, page, pageRelation, pageContent;
+
+	try {
+		response = await fetch(`${process.env.CMS_API}/menus/1${query}`, {
+			// headers: {
+			//     authorization: `Bearer ${process.env.CMS_TOKEN}`,
+			// },
+		});
+
+		if (!response.ok) {
+			throw new Error(`Failed to fetch menu data: ${response.statusText}`);
+		}
+
+		data = await response.json();
+	} catch (error) {
+		return { error: "Failed to fetch menu data" };
+	}
+
+	try {
+		page = data.data.attributes.items.data.find((item: { attributes: { url: string } }) => item.attributes.url.endsWith(slug[0]));
```

Comment:
That seems veeeeery error-prone and I don't think that a try...catch block can catch a „cannot read property x of undefined“. You have to at least add a ? behind each property call.

```suggestion
		page = data?.data?.attributes?.items?.data?.find((item: { attributes: { url: string } }) => item?.attributes?.url?.endsWith(slug[0]));
```

Reply:
Done